Remove hanging files

chetanyagoyal opened the pull request:
This PR attempts to remove hanging files (like ./openfasoc/generators/temp-sense-gen/tools/verify_op.sh) and includes their functionalities in the corresponding report parse files 
Also does the same for the parse_rpt.py file found in the .github/scripts section to account for this functionality in the CI

chetanyagoyal commented:
@saicharan0112 @msaligane

chetanyagoyal commented:
@saicharan0112 As the ldo-gen tools folder does not have a parse report file, would you suggest that I make one along the lines of what the .github/scripts parse_rpt.py file does?

saicharan0112 commented:
> @saicharan0112 As the ldo-gen tools folder does not have a parse report file, would you suggest that I make one along the lines of what the .github/scripts parse_rpt.py file does?

I think its a good idea to have all generators follow the same format to check for drc and lvs errors. tbh I would expect a function inside the main script itself but for now it is ok to have a different file. 

Here is a suggestion.. can you list all files that are generated by the three generators (cryo-gen, ldo-gen and temp-sense-gen) and see if the existence of these files are checked along with the drc and lvs errors please?

chetanyagoyal commented:
@saicharan0112 are there any other changes required before this is merged?

saicharan0112 commented on openfasoc/generators/temp-sense-gen/tools/parse_rpt.py:
```diff
@@ -16,3 +19,26 @@
+    raise ValueError("work directory does not exist!")
+else:
+    filename = work_dir + module_name
+    for file in (filename + ".gds", filename + ".spice", filename + ".v", filename + ".def", filename + "_pex.spice", filename + ".sdc"):
```
Avoid hard-coding inside the logic itself. I would suggest to use list here to store the file extensions so that in future it can be updated with more file extensions if needed. Refer to how the drc report and lvs report file name are stored in the very beginning.

chetanyagoyal replied:
Resolved in [https://github.com/idea-fasoc/OpenFASOC/pull/251/commits/f3057569c854d497f2a7f01566381986b3810d44](url)

saicharan0112 commented on .github/scripts/parse_rpt.py:
```diff
@@ -1,33 +1,74 @@
+        raise ValueError("work directory does not exist!")
+    else:
+        filename = work_dir + module_name
+        for file in (filename + ".gds", filename + ".spice", filename + ".v", filename + ".def", filename + "_pex.spice", filename + ".sdc"):
```
Please refer to my last comment on this same thing.

saicharan0112 reviewed (changes requested):
There is a lot of code redundancy. I think you can have one single function for these checks (probably you can use the common directory outside the generators?) and reuse with generator specific files to do checks.

saicharan0112 commented on .github/scripts/parse_rpt.py:
```diff
@@ -1,33 +1,74 @@
+    	data = json.load(file)
+    print('Found .json config file...')
+
+    module_name = data.get("module_name", "default")
```
this part will fail if there is not `test.json` file. Please check the logic again under the else part here.

chetanyagoyal commented:
> There is a lot of code redundancy. I think you can have one single function for these checks (probably you can use the common directory outside the generators?) and reuse with generator specific files to do checks.

@saicharan0112 Fixed all issues, please check

saicharan0112 commented on openfasoc/generators/cryo-gen/tools/parse_rpt.py:
```diff
@@ -1,16 +1,19 @@
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', '..'))
+from common.check_gen_files import check_gen_files
+
+drc_filename = "flow/reports/sky130hd/cryo/6_final_drc.rpt"
```
cryo-gen supports three libraries of sky130 - sky130hd, sky130hs, sky130hvl. Can you please update so that all three variants are checked?

saicharan0112 commented on openfasoc/generators/cryo-gen/tools/parse_rpt.py:
```diff
@@ -1,16 +1,19 @@
     print("DRC is clean!")
 
-lvs_filename = "flow/reports/sky130hd/cyro/6_final_lvs.rpt"
+lvs_filename = "flow/reports/sky130hd/cryo/6_final_lvs.rpt"
```
same here as well...

saicharan0112 commented on openfasoc/generators/cryo-gen/tools/parse_rpt.py:
```diff
@@ -22,3 +25,13 @@
+json_filename = "test.json"
+
+if os.path.exists(json_filename):
+    if check_gen_files():
```
I think it would be good to pass the `json_file_name` as an argument to `check_gen_files()` so that you can avoid changing this filename at two places (at the generator side as well as inside the `common/check_gen_files.py`.

saicharan0112 commented on openfasoc/generators/temp-sense-gen/tools/parse_rpt.py:
```diff
@@ -16,3 +21,13 @@
+json_filename = "test.json"
+
+if os.path.exists(json_filename):
+    if check_gen_files():
```
same here as well...

saicharan0112 reviewed (changes requested):
@chetanyagoyal The updates are much better now, thanks for taking the suggestions. I added my comments for your updates. Let me know once you push the changes.

saicharan0112 commented on .github/scripts/parse_rpt.py:
```diff
@@ -1,33 +1,66 @@
+        else:
+            print("LVS is clean!")
+
+if ((len(sys.argv) > 1) and ((sys.argv[1] == "sky130hvl_ldo") or (sys.argv[1] == "sky130hvl_ldo_full"))) or ((len(sys.argv) > 1) and (sys.argv[1] == "sky130hd_cryo")):
```
@chetanyagoyal : can you please explain what you are trying to do here? I don't understand what you are expecting as arguments to this script.

chetanyagoyal replied:
Sure! Its based off the way parse_rpt.py is being called in the respective yaml files 
For ex. - if its `python3 parse_rpt.py sky130hvl_ldo`, I want this statement to be triggered and argv[1] here will be sky130hvl_ldo.

saicharan0112 replied:
ok but why you don't want to check the generated files using `check_gen_files()` if these (`sky130hvl_ldo`, `sky130hvl_ldo_full`, `sky130hd_cryo` are the arguments, and consider directly that the flow is clean?

chetanyagoyal replied:
Oh right, I have already removed this statement. I meant to do that earlier but it slipped my mind. These generators will also be checked with the new script

saicharan0112 commented:
FYI, the test.json file is already checked by the generator itself. without
that, the generator wouldn't start in the first place. I am not sure why
you are considering that file to check for the expected generated files for
each generator.

On Wed, Nov 8, 2023 at 10:17 PM Chetanya ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .github/scripts/parse_rpt.py
> <https://github.com/idea-fasoc/OpenFASOC/pull/251#discussion_r1386923241>:
>
> >          raise ValueError("LVS failed!")
>      else:
>          print("LVS is clean!")
> +else:
> +    with open(lvs_filename) as f:
> +        f1 = f.read()
> +
> +        if "failed" in f1:
> +            raise ValueError("LVS failed!")
> +        else:
> +            print("LVS is clean!")
> +
> +if ((len(sys.argv) > 1) and ((sys.argv[1] == "sky130hvl_ldo") or (sys.argv[1] == "sky130hvl_ldo_full"))) or ((len(sys.argv) > 1) and (sys.argv[1] == "sky130hd_cryo")):
>
> Oh right, I have already removed this statement. I meant to do that
> earlier but it slipped my mind. These generators will also be checked with
> the new script
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idea-fasoc/OpenFASOC/pull/251#discussion_r1386923241>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYCOBKOYFVGNVETRGAI2FOLYDOZQTAVCNFSM6AAAAAA53NNZ7OVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMRQHAZDINZSG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

saicharan0112 reviewed (changes requested):
@chetanyagoyal please look at my comments on cryo-gen library variants. Please refer to the cryo-gen.yml file on how the cryo generator is tested.

saicharan0112 commented on .github/scripts/parse_rpt.py:
```diff
@@ -1,33 +1,62 @@
-
-    if "failed" in f1:
+# cryo LVS check 
+if (len(sys.argv) > 1) and (sys.argv[1] == str(lib[0])):
```
@chetanyagoyal : can you explain more about this please? what is the difference between the checks that is done in this `if` block and the `else` block. looks like they both are using the same `lvs_filename`?

chetanyagoyal replied:
I think this one is required because the cryo's parse report file checks the drc and lvs in `flow/reports`. Seems like the work directory doesn't exist for it, which is why i added the if statement

the `lib = os.listdir(..)` statement checks which cryo library was used to go one level deeper into `flow/reports`

saicharan0112 replied:
that's alright but you are checking the file with the same variable. the code under `if` and `else` logically does the same thing ?

chetanyagoyal replied:
Yes i've fixed this now 
If checks for temp-sense and ldo, and the elif checks for the various possible cryo-gens

saicharan0112 commented on .github/scripts/parse_rpt.py:
```diff
@@ -1,33 +1,62 @@
+lib = os.listdir(dir_path)
+
+if (len(sys.argv) > 1) and (sys.argv[1] == str(lib[0])):
+    drc_filename = "flow/reports/" + sys.argv[1] + "/cryo/6_final_drc.rpt"
```
@chetanyagoyal : I am not sure about this tbh. The `if` block and the last `else` does the same thing. I don't know why you are considering the drc report from the `flow/reports/...` path since the reports will be copied to the `work` folder anyways in the end. If you consider that, they you don't need the first `if` and the last `else` should take care of the cryo-gen reports check

saicharan0112 commented on openfasoc/generators/PMU-gen/tools/parse_rpt.py:
```diff
@@ -22,3 +22,26 @@
 else:
     print("LVS is clean!")
+
+# with open('test.json', 'r') as file:
```
can we remove this comments part please? I don't know when this generator work will be resumed and this might confuse people who will be developing this in future. maybe we can keep this file like it was ?

saicharan0112 commented on openfasoc/generators/common/__init__.py:
```diff
@@ -7,6 +7,9 @@
-
+- `common.check_gen_files`
+	1. `check_gen_files(parameters: json_filename, is_tempsense)`: Used to check if the various files that should be generated by the flow are present in their required directories. 
+- `common.check_gen_extensions`
```
I don't see any python file with this name in this directory. Can you please more about this?

chetanyagoyal replied:
the check_gen_files.py file exists in the `generators/common` directory, and it has the check_gen_files function
Each generator (temp-sense, ldo-gen, cryo) has check_gen_extensions files in their `tools` directories which contains the extensions of the files generated by the flow

saicharan0112 replied:
right but why are you adding that function as `common.check_gen_extensions`. That doesn't belong to the common library, does it?

saicharan0112 commented on openfasoc/generators/dcdc-gen/tools/parse_rpt.py:
```diff
@@ -22,3 +22,27 @@
     print("LVS is clean!")
+
+# with open('test.json', 'r') as file:
+#     data = json.load(file)
```
same here as well...

saicharan0112 commented on openfasoc/generators/scpa-gen/tools/parse_rpt.py:
```diff
@@ -19,3 +19,27 @@
 # 	print("LVS is clean!")
+
+# with open('test.json', 'r') as file:
+#     data = json.load(file)
```
same here as well...

saicharan0112 commented on openfasoc/generators/temp-sense-gen/tools/parse_rpt.py:
```diff
@@ -1,3 +1,8 @@
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', '..'))
+from common.check_gen_files import check_gen_files
+
 drc_filename = "flow/reports/sky130hd/tempsense/6_final_drc.rpt"
```
TBH I don't know why we are considering the lvs and drc reports from this folder since we already have them in the work directory. Can we change this and keep everything aligned to a single convention?

saicharan0112 commented on .github/scripts/parse_rpt.py:
```diff
@@ -1,33 +1,62 @@
 import sys
```
Can you please write a brief summary of how this script just for the documentation purposes? it would be useful if someone wants to add more functionalities and doesn't want to break this script.

saicharan0112 reviewed (changes requested):
Hope you are working on logic summary in the beginning of `parse_rpt.py` under `.github/scripts` and `common/check_gen_files.py`

chetanyagoyal commented:
> Hope you are working on logic summary in the beginning of `parse_rpt.py` under `.github/scripts` and `common/check_gen_files.py`

Addressed all issues, please check

saicharan0112 reviewed (approved):
Thanks @chetanyagoyal . It can still be improved but it looks very good for now. Right now cryo-gen and ldo are not reaching till the stage where the reports are checked and the outputs are verified, so let's see how it goes once they are active.
